connectors/source.py

```diff
@@ -107,6 +132,114 @@ def is_empty(self):
     def to_dict(self):
         return dict(self._raw_config)
 
+    def is_valid(self):
+        """Validates every Field against its `validations`.
+
+        Raises ConfigurableFieldValueError if any validation errors are found,
```

Comment:
signature nit: if the function raises on validation error, then we don't need it to return `True` as we probably won't read that value when it does not raise.

Reply:
also, if we just raise, I'd rename it to `check_valid` as `is_valid` implies it returns `True` or `False`